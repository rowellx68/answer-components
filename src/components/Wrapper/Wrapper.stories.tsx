import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import Wrapper, { WrapperProps } from './Wrapper'

export default {
  title: 'Components/Layout/Wrapper',
  component: Wrapper,
  argTypes: {
    className: { control: 'text' },
  },
} as Meta

const Template: Story<WrapperProps> = (args) => (
  <div className="bg-yellow-100">
    <Wrapper {...args}>
      <h2 className="text-2xl md:text-3xl font-extralight">
        This is content is inside a Wrapper.
      </h2>
    </Wrapper>
    <p className="py-10">This one is not.</p>
  </div>
)

export const BasicUsage = Template.bind({})
BasicUsage.args = {
  className: 'p-6 bg-gray-200',
}
