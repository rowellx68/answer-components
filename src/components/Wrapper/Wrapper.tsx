import React from 'react'

export interface WrapperProps {
  className?: string
}

const Wrapper: React.FC<WrapperProps> = ({ children, className = '' }) => {
  return (
    <div className={`container mx-auto px-4 sm:px-6 lg:px-8 ${className}`}>
      {children}
    </div>
  )
}

export default Wrapper
