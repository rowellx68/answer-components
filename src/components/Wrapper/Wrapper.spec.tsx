import React from 'react'
import renderer from 'react-test-renderer'

import Wrapper from './Wrapper'

test('renders Wrapper correctly', () => {
  const element = renderer
    .create(<Wrapper>Content goes here.</Wrapper>)
    .toJSON()

  expect(element).toMatchSnapshot()
})

test('renders Wrapper correctly when passing className', () => {
  const element = renderer
    .create(<Wrapper className="bg-gray-200">Content goes here.</Wrapper>)
    .toJSON()

  expect(element).toMatchSnapshot()
})
