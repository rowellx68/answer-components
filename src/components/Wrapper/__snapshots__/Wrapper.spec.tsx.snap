// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Wrapper correctly 1`] = `
<div
  className="container mx-auto px-4 sm:px-6 lg:px-8 "
>
  Content goes here.
</div>
`;

exports[`renders Wrapper correctly when passing className 1`] = `
<div
  className="container mx-auto px-4 sm:px-6 lg:px-8 bg-gray-200"
>
  Content goes here.
</div>
`;
